### ISSUE TITLE
azurerm_windows_virtual_machine identity_ids are case-sensitive, and shouldn't be

### ISSUE THREAD

**Issue by arestarh** [terraform-providers/terraform-provider-azurerm#10099]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform **v0.14.3**
AzureRM Provider **v2.41.0**

### Affected Resource(s)

* `azurerm_windows_virtual_machine`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Debug Output
Observe the difference in case of "resourcegroups" vs. "resourceGroups" below in the identity_ids.
```
~ resource "azurerm_windows_virtual_machine" "my_vm" {
2021-01-06T14:16:23.5207447Z         id                         = "/subscriptions/***/resourceGroups/my-rg/providers/Microsoft.Compute/virtualMachines/my-vm"
2021-01-06T14:16:23.5208123Z         name                       = "my-vm"
2021-01-06T14:16:23.5208490Z         tags                       = {
2021-01-06T14:16:23.5209272Z             "barometer_id"   = "tag1"
2021-01-06T14:16:23.5209688Z             "owner"          = "tag2"
2021-01-06T14:16:23.5214872Z             "project"        = "tag3"
2021-01-06T14:16:23.5215270Z             "wk_class_Patch" = "tag4"
2021-01-06T14:16:23.5215905Z         }
2021-01-06T14:16:23.5216152Z         # (21 unchanged attributes hidden)
2021-01-06T14:16:23.5216326Z 
2021-01-06T14:16:23.5216448Z 
2021-01-06T14:16:23.5216615Z       ~ identity {
2021-01-06T14:16:23.5216844Z           ~ identity_ids = [
2021-01-06T14:16:23.5217398Z               + "/subscriptions/***/resourceGroups/my-rg1/providers/Microsoft.ManagedIdentity/userAssignedIdentities/my-uami",
2021-01-06T14:16:23.5218665Z               - "/subscriptions/***/resourcegroups/my-rg1/providers/Microsoft.ManagedIdentity/userAssignedIdentities/my-uami",
2021-01-06T14:16:23.5219513Z                 # (1 unchanged element hidden)
2021-01-06T14:16:23.5219762Z             ]
2021-01-06T14:16:23.5222347Z             # (1 unchanged attribute hidden)
2021-01-06T14:16:23.5223373Z         }
2021-01-06T14:16:23.5224046Z 
2021-01-06T14:16:23.5224273Z         # (2 unchanged blocks hidden)
2021-01-06T14:16:23.5224494Z     }
```

### Expected Behaviour
The property is identical other than case, so resource does not need to be updated and should not be identified by terraform plan for an update.

### Actual Behaviour
Terraform wants to update the resource

### Steps to Reproduce

1. `terraform apply`

### Important Factoids
This occurs every run, even after applying. I run terraform apply again, and EVERY TIME it thinks it needs to update the resource.

### References

* #0000

**Comment by arestarh** [comment]
It's pretty the same issue described in https://github.com/terraform-providers/terraform-provider-azurerm/issues/10029

**Comment by jbinko** [comment]
@katbyte : I guess can be closed now

**Comment by tombuildsstuff** [comment]
Duplicate of #10300 - fixed in #10722 - shipping in v2.50.0 - closing as above.
